### ISSUE TITLE
Add support for ABAR Diaper

### ISSUE THREAD

**Issue by DoomRater** [DoomRater/DrizzleScript#39]
This is a funny one.  ABAR diapers don't have the main shell as the main prim.  I'd like to ask the creator of the ABAR diaper to relink the set so that the main prim is the shell of the diaper, which will reduce the amount of code editing that is needed to support a diaper.  If for whatever reason this solution isn't feasible, DrizzleScript already supports having a prim that isn't the root prim, but the variables that need to be edited aren't exposed to users at any level.  That means people have to edit the scripts just to get their diapers working with DrizzleScript.

This raises a few questions.  First, should these variables be exposed in some sort of developer menu?  Is there enough overhead to even implement something like that?  And where should I put documentation on how to get things working with DrizzleScript?  It's clear that there are people who want to put these scripts into other diapers than the default, and the scripts are such that only people familiar with how they work are going to be able to edit the scripts in such a fashion.

Adding support for ABAR Sculpt should be relatively simple.  Thing is... automatically detecting the ABAR diaper may not be possible the way I have attempted to implement auto-detection.  Not only are there two, but I am not entirely sure the creator of the prim will stay the same as it's being developed.  Also, the mesh version's mapping is different enough where I imagine custom skins will have to be made for it.

In summary, supporting the mesh ABAR diaper is a matter of making sure the correct prim is detected in code, as well as implementing the menu written for Kawaii diapers so that the ruffles are supported from the same menu.  But the other questions that have arisen from speaking with the creator need to be addressed, especially the lack of documentation of what the script assumes and what each of the variables do.

**Comment by DoomRater** [comment]
One idea I have off the top of my head is to move the code that handles application of textures to an outside script where it can be more easily spotted.  I think the cons of this idea are pretty obvious- more scripts to worry about.  But it could potentially clean up a bunch of code and not require tracking and finding the main prim in two different scripts.

**Comment by DoomRater** [comment]
One more thing I can note about ABAR: the marketplace listing in question shows me that the creator of the original objects is not the person who wants to have it scripted for the diaper.  The requestor for support will have to make sure that the diaper is not released with permissions that go against the license for the object.  They will also need to make sure the DrizzleScript code is kept full permissions according to the RPL license.  However it also tells me that I can support the original marketplace listing by checking the creator of the object, which requires less code overhead than anticipated.

**Comment by DoomRater** [comment]
Support for their mesh offering to end up top priority.

**Comment by DoomRater** [comment]
ABAR Sculpt support is still in, but I cannot auto-detect it due to the likelihood of the owner of the prim changing, and the fact that the ABAR Mesh diaper has the same creator. I am going to need some sort of way to tell the diaper what type of diaper it is in without rewriting code all the time.  And that is likely to happen by way of an included Notecard.

Also to note that I cannot find good texture settings for the ABAR Mesh diaper.  It is likely that someone who wants to make this diaper available for others will need to design their own textures.  Making textures is not my forte, so I will leave this task to any others.
